pettingzoo/utils/save_observation.py

```diff
@@ -39,7 +48,8 @@ def save_observation(env, agent=None, all_agents=False, save_dir=os.getcwd()):
         os.makedirs(save_folder, exist_ok=True)
 
         observation = env.observe(a)
-        rescaled = observation.astype(np.uint8)
-        im = Image.fromarray(rescaled)
-        fname = os.path.join(save_folder, str(a) + ".png")
-        im.save(fname)
+        if observation is not None:
```

Comment:
Is `observation is None` a valid value in this context? My intuition is that it should never happen, so it should raise an error (as it would right now when you try to do `None.astype`)

With this change, this error would just quietly be ignored

Reply:
I added the check because pyright complained about using `.astype` on a None value.

Other possibilities are:
* ignore the error with `# pyright: ignore`.
* check if observation is None and if so raise an error.

Let me know what you prefer

Reply:
Added an assertion error if observation is None